controllers/storagecluster/ocs_operator_config.go

```diff
@@ -102,14 +102,27 @@ func (r *StorageClusterReconciler) getClusterID() string {
 	return fmt.Sprint(clusterVersion.Spec.ClusterID)
 }
 
-// getCephFSKernelMountOptions returns the kernel mount options for cephfs
+// getCephFSKernelMountOptions returns the kernel mount options for CephFS based on the spec on the StorageCluster
 func getCephFSKernelMountOptions(sc *ocsv1.StorageCluster) string {
-	// If it is an provider or consumer/external cluster, we don't need to set any mount options
+	// If Encryption is enabled, Always use secure mode
+	if sc.Spec.Network != nil && sc.Spec.Network.Connections != nil &&
+		sc.Spec.Network.Connections.Encryption.Enabled {
+		return "ms_mode=secure"
+	}
+
+	// If Encryption is not enabled, but Compression or RequireMsgr2 is enabled, use prefer-crc mode
+	if sc.Spec.Network != nil && sc.Spec.Network.Connections != nil &&
+		((sc.Spec.Network.Connections.Compression != nil && sc.Spec.Network.Connections.Compression.Enabled) ||
+			sc.Spec.Network.Connections.RequireMsgr2) {
+		return "ms_mode=prefer-crc"
+	}
+
+	// Non of Encryption, Compression, RequireMsgr2 are enabled on the StorageCluster
```

Comment:
`Non -> None ` We also need to correct it in the commit-msg. Also, let's add that network spec has higher precedence even in the External or Provider cluster. So that we don't move this if block above in the future.

Reply:
Sure

Reply:
Done